### PR TITLE
Move discussion reply box (and convert to ES6)

### LINKED ISSUE #2973: Resizing the notes and message board thread Talk  comment box so that it fits right below the previous comments
A version of this ticket was closed previously  with the response that a Talk design pass was being made and would address this issue. That was a while ago, so I've created a new ticket. 

example: hit reply in a [thread](https://www.zooniverse.org/projects/mschwamb/planet-four-terrains/talk/29/86792) like this

Full page capture when you hit reply and there are not enough comments to fill all the way down to the bottom edge of the left side bar:

fireshot capture 25 - planet four_ t__ - https___www zooniverse org_project

What you see in your visible browser when you hit reply to the comment:

screen shot 2016-09-05 at 4 14 53 pm

same thing for: https://www.zooniverse.org/projects/mschwamb/planet-four-terrains/talk/10/85980?page=1&scrollToLastComment=true

screen shot 2016-09-05 at 4 25 28 pm

If there is any developer time in the future to resize the comment box so it fits in the space below the already published comments, I think it would be very welcomed by the Talk community. Replying on Talk thread is a commonly used feature, so it would be nice if one could see the full discussion or subject image rather than having this huge gap

This behavior happens in all browsers I tested (Chrome, fixfox, and safari for mac) 

cc @parrish

### PR DESCRIPTION
**Pull request by eatyourgreens**
Fixes #2973 .

Describe your changes.
Rewrites the discussion thread comment box in ES6 and moves it from the very bottom of the page to the bottom of the list of comments.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://fix-2973.pfe-preview.zooniverse.org/?env=production
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/talk/discussion-comment.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            reply={this.props.reply}
+            logSubmit={true}
+            onClickClearReply={this.props.onClearReply}
+            header={null}
+          />
+        </section>
+      );
+    } else {
+      return (
+        <p>Please <button className="link-style" type="button" onClick={this.promptToSignIn}>sign in</button> to contribute to the discussion</p>
```
Comment (eatyourgreens) [inline app/talk/discussion-comment.jsx]: `this.promptToSignIn` is undefined here.

File: app/talk/discussion-comment.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  commentValidations(commentBody) {
+    // TODO: return true if any additional validations fail
+    const commentValidationErrors = getErrors(commentBody, commentValidations);
+    this.setState({ commentValidationErrors });
+    return !!commentValidationErrors.length;
+  }
+
+  render() {
+    if (!!this.props.discussion && this.props.discussion.locked) {
+      return this.lockedMessage();
```
Comment (eatyourgreens) [inline app/talk/discussion-comment.jsx]: `this.lockedMessage` is undefined here.

This if block should probably be moved up into the parent component.

**Comment by eatyourgreens** [comment]
Doesn't work for locked threads or anonymous users yet.

File: app/talk/discussion.cjsx [hunk 3]
```diff
@@ -49,6 +46,18 @@ module.exports = React.createClass
     if @props.params.discussion isnt nextProps.params.discussion
       @setDiscussion(nextProps.params.discussion)
         .then => @setComments(nextProps.location.query.page ? 1)
+        .then =>
+          if nextProps.user isnt @props.user
+            talkClient
```
Comment (srallen) [inline app/talk/discussion.cjsx]: I'm not really familiar how the talkClient handles errors, should there be a catch on this?

File: app/talk/discussion.cjsx [hunk 4]
```diff
@@ -69,18 +78,18 @@ module.exports = React.createClass
             pathname: @props.location.pathname
             query: @props.location.query
 
+        if @props.user?
+          talkClient
```
Comment (srallen) [inline app/talk/discussion.cjsx]: Same question about error catching

File: app/talk/discussion.cjsx (hunk 3 above)
Comment (eatyourgreens) [inline app/talk/discussion.cjsx]: I'm not sure. I copied this straight from the coffeescript without really thinking about it.

Comment (eatyourgreens) [inline app/talk/discussion.cjsx]: Probably should be catching errors, though. If the roles call fails, the only side effect I can think of is that you don't get the banner under your username beside the comment box.

**Pull request merged**